test/minimum_eigensolvers/test_qaoa.py

```diff
@@ -290,8 +290,11 @@ def _sample_most_likely(self, state_vector: QuasiDistribution) -> np.ndarray:
         Returns:
             Binary string as numpy.ndarray of ints.
         """
-        values = list(state_vector.values())
-        n = int(np.log2(len(values)))
+        max_key = max(state_vector.keys())
```

Comment:
The previous version contains a bug where some samples are missing, e.g., `{0: 0.1, 1:0.4, 3:0.5}` (`2` is missing).